Sources/TuistKit/Services/EditService.swift

```diff
@@ -56,27 +60,29 @@ final class EditService {
         let plugins = await loadPlugins(at: path)
 
         if !permanent {
-            try withTemporaryDirectory(removeTreeOnDeinit: true) { generationDirectory in
-                EditService.temporaryDirectory = generationDirectory
-
-                signalHandler.trap { _ in
-                    try? EditService.temporaryDirectory.map(FileHandler.shared.delete)
-                    exit(0)
-                }
-
-                guard let selectedXcode = try XcodeController.shared.selected() else {
-                    throw EditServiceError.xcodeNotSelected
-                }
-
-                let workspacePath = try projectEditor.edit(
-                    at: path,
-                    in: generationDirectory,
-                    onlyCurrentDirectory: onlyCurrentDirectory,
-                    plugins: plugins
-                )
-                logger.pretty("Opening Xcode to edit the project. Press \(.keystroke("CTRL + C")) once you are done editing")
-                try opener.open(path: workspacePath, application: selectedXcode.path, wait: true)
+            let cacheDirectoryProvider = try cacheDirectoryProviderFactory.cacheDirectories()
+            let cacheDirectory = try cacheDirectoryProvider.tuistCacheDirectory(for: .editProjects)
+            let cachedManifestDirectory = cacheDirectory.appending(component: path.pathString.md5)
+            EditService.temporaryDirectory = cachedManifestDirectory
+            
+            signalHandler.trap { _ in
+                try? EditService.temporaryDirectory.map(FileHandler.shared.delete)
+                exit(0)
             }
+            
+            guard let selectedXcode = try XcodeController.shared.selected() else {
+                throw EditServiceError.xcodeNotSelected
+            }
+            
+            let workspacePath = try projectEditor.edit(
+                at: path,
+                in: cachedManifestDirectory,
+                onlyCurrentDirectory: onlyCurrentDirectory,
+                plugins: plugins
+            )
+            logger.pretty("Opening Xcode to edit the project. Press \(.keystroke("CTRL + C")) once you are done editing")
+            try opener.open(path: workspacePath, application: selectedXcode.path, wait: true)
```

Comment:
@waltflanagan since you are touching this part of the project 😬, what do you think about adjusting `opener`'s logic, to detect when the underlying process terminates (i.e., Xcode is closed), to continue the execution of the program instead of forcing the user to send a SIGKILl signal with CTRL+C? In theory, it should be possible with [this](https://developer.apple.com/documentation/foundation/process/1408746-terminationhandler)